pkg/controller/manager.go

```diff
@@ -75,8 +75,10 @@ func (m *Manager) updateController(name string, params ControllerParams) *manage
 
 		// Notify the goroutine of the params update.
 		select {
-		case ctrl.update <- ctrl.params:
-		default:
+		case ctrl.update <- ctrl.params: // Send if empty
+		default: // Overwrite if full
```

Comment:
What guarantees do we have that ctrl.update is not emptied, on a different go routine, which will then block on line 80 forever?

Reply:
Hmm, that's a good point, the `updateController` is wrapped in a mutex, so that one should be safe, but the `runController`'s select statement is not.  Do you see a way to tackle this concern with this approach? Or you think we should update the consumers of this controller to stop relying on the params sent and just use it as a notifier?

Reply:
IMO I think 

>  you think we should update the consumers of this controller to stop relying on the params sent and just use it as a notifier?

is a better option. 

Trying to massage the controller code to workaround an esoteric case in Cilium seems like we are adding additional fragility to already fragile code.